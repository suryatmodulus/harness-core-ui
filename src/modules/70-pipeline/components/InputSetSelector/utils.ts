import type { IconName, SelectOption } from '@harness/uicore'
import type { EntityGitDetails, InputSetSummaryResponse } from 'services/pipeline-ng'
import css from './InputSetSelector.module.scss'

type InputSetLocal = InputSetSummaryResponse & SelectOption
export interface InputSetValue extends InputSetLocal {
  type: InputSetSummaryResponse['inputSetType']
  gitDetails?: EntityGitDetails
}

export const getIconByType = (type: InputSetSummaryResponse['inputSetType']): IconName => {
  return type === 'OVERLAY_INPUT_SET' ? 'step-group' : 'yaml-builder-input-sets'
}

export const onDragStart = (event: React.DragEvent<HTMLLIElement>, row: InputSetValue): void => {
  event.dataTransfer.setData('data', JSON.stringify(row))
  event.currentTarget.classList.add(css.dragging)
}

export const onDragEnd = (event: React.DragEvent<HTMLLIElement>): void => {
  event.currentTarget.classList.remove(css.dragging)
}

export const onDragLeave = (event: React.DragEvent<HTMLLIElement>): void => {
  event.currentTarget.classList.remove(css.dragOver)
}

export const onDragOver = (event: React.DragEvent<HTMLLIElement>): void => {
  if (event.preventDefault) {
    event.preventDefault()
  }
  event.currentTarget.classList.add(css.dragOver)
  event.dataTransfer.dropEffect = 'move'
}
