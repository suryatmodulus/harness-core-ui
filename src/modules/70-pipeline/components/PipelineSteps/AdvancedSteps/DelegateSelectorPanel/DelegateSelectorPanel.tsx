/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import type { FormikProps } from 'formik'
import { useParams } from 'react-router-dom'

import MultiTypeDelegateSelector from '@common/components/MultiTypeDelegateSelector/MultiTypeDelegateSelector'
import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import { useVariablesExpression } from '@pipeline/components/PipelineStudio/PiplineHooks/useVariablesExpression'

export interface DelegatePanelProps {
  formikProps: FormikProps<{
    delegateSelectors?: string[]
  }>
  isReadonly: boolean
}

export default function DelegateSelectorPanel(props: DelegatePanelProps): React.ReactElement {
  const { isReadonly } = props
  const { projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
  const { expressions } = useVariablesExpression()

  return (
    <MultiTypeDelegateSelector
      name="delegateSelectors"
      disabled={isReadonly}
      inputProps={{ projectIdentifier, orgIdentifier }}
      expressions={expressions}
    />
  )
}
