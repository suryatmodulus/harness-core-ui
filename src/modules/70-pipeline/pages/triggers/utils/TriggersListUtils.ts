/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@wings-software/uicore'
// temporary mock data
import { parse } from 'yaml'
import type { AddDrawerMapInterface } from '@common/components/AddDrawer/AddDrawer'
import type { StringKeys } from 'framework/strings'
import { manifestTypeIcons, ManifestDataType } from '@pipeline/components/ManifestSelection/Manifesthelper'
import { ArtifactIconByType, ENABLED_ARTIFACT_TYPES } from '@pipeline/components/ArtifactsSelection/ArtifactHelper'
import { TriggerTypes, AWS_CODECOMMIT, AwsCodeCommit } from './TriggersWizardPageUtils'

export const GitSourceProviders: Record<string, { value: string; iconName: IconName }> = {
  GITHUB: { value: 'Github', iconName: 'github' },
  GITLAB: { value: 'Gitlab', iconName: 'service-gotlab' },
  BITBUCKET: { value: 'Bitbucket', iconName: 'bitbucket-selected' },
  AWS_CODECOMMIT: { value: 'AwsCodeCommit', iconName: 'service-aws-code-deploy' },
  CUSTOM: { value: 'Custom', iconName: 'build' }
}

const TriggerTypeIcons = {
  SCHEDULE: 'trigger-schedule',
  NEW_ARTIFACT: 'new-artifact'
}
export const getTriggerIcon = ({
  type,
  webhookSourceRepo,
  buildType
}: {
  type: string
  webhookSourceRepo?: string // string temporary until backend
  buildType?: string
}): IconName => {
  const updatedWebhookSourceRepo =
    webhookSourceRepo === AwsCodeCommit ? AWS_CODECOMMIT : webhookSourceRepo?.toUpperCase()
  const webhookSourceRepoIconName =
    webhookSourceRepo && updatedWebhookSourceRepo && GitSourceProviders[updatedWebhookSourceRepo]?.iconName
  if (type === TriggerTypes.WEBHOOK && webhookSourceRepoIconName) {
    return webhookSourceRepoIconName as IconName
  } else if (type === TriggerTypes.SCHEDULE) {
    return TriggerTypeIcons.SCHEDULE as IconName
  } else if (type === TriggerTypes.MANIFEST && buildType) {
    if (buildType === ManifestDataType.HelmChart) {
      return manifestTypeIcons.HelmChart
    }
  } else if (type === TriggerTypes.ARTIFACT && buildType) {
    switch (buildType) {
      case ENABLED_ARTIFACT_TYPES.Gcr:
        return ArtifactIconByType.Gcr
      case ENABLED_ARTIFACT_TYPES.Ecr:
        return ArtifactIconByType.Ecr
      case ENABLED_ARTIFACT_TYPES.DockerRegistry:
        return ArtifactIconByType.DockerRegistry
    }
  }
  return 'yaml-builder-trigger'
}

const triggerDrawerMap = (getString: (key: StringKeys) => string): AddDrawerMapInterface => ({
  drawerLabel: getString('pipeline.triggers.triggersLabel'),
  showAllLabel: getString('pipeline.triggers.showAllTriggers'),
  categories: [
    {
      categoryLabel: getString('execution.triggerType.WEBHOOK'),
      categoryValue: 'Webhook',
      items: [
        {
          itemLabel: getString('common.repo_provider.githubLabel'),
          value: GitSourceProviders.GITHUB.value,
          iconName: GitSourceProviders.GITHUB.iconName
        },
        {
          itemLabel: getString('common.repo_provider.gitlabLabel'),
          value: GitSourceProviders.GITLAB.value,
          iconName: GitSourceProviders.GITLAB.iconName
        },
        {
          itemLabel: getString('common.repo_provider.bitbucketLabel'),
          value: GitSourceProviders.BITBUCKET.value,
          iconName: GitSourceProviders.BITBUCKET.iconName
        },
        {
          itemLabel: getString('common.repo_provider.awscodecommit'),
          value: GitSourceProviders.AWS_CODECOMMIT.value,
          iconName: GitSourceProviders.AWS_CODECOMMIT.iconName
        },
        {
          itemLabel: getString('common.repo_provider.customLabel'),
          value: GitSourceProviders.CUSTOM.value,
          iconName: GitSourceProviders.CUSTOM.iconName
        }
      ]
    },
    {
      categoryLabel: getString('pipeline.triggers.scheduledLabel'),
      categoryValue: 'Scheduled',
      items: [
        {
          itemLabel: getString('pipeline.triggers.cronLabel'),
          value: 'Cron',
          iconName: TriggerTypeIcons.SCHEDULE as IconName
        }
      ]
    }
  ]
})

export const getSourceRepoOptions = (getString: (str: StringKeys) => string): { label: string; value: string }[] => [
  { label: getString('common.repo_provider.githubLabel'), value: GitSourceProviders.GITHUB.value },
  { label: getString('common.repo_provider.gitlabLabel'), value: GitSourceProviders.GITLAB.value },
  { label: getString('common.repo_provider.bitbucketLabel'), value: GitSourceProviders.BITBUCKET.value },
  { label: getString('common.repo_provider.codecommit'), value: GitSourceProviders.AWS_CODECOMMIT.value },
  { label: getString('common.repo_provider.customLabel'), value: GitSourceProviders.CUSTOM.value }
]

export const getCategoryItems = (getString: (key: StringKeys) => string): AddDrawerMapInterface =>
  triggerDrawerMap(getString)

export interface ItemInterface {
  itemLabel: string
  iconName: IconName
  value: string
  visible?: boolean
  disabled?: boolean
  categoryValue?: string
}

export interface TriggerDataInterface {
  triggerType: string
  sourceRepo?: string
  manifestType?: string
  artifactType?: string
}

export const getEnabledStatusTriggerValues = ({
  data,
  enabled,
  getString
}: {
  data: any
  enabled: boolean
  getString: (key: StringKeys) => string
}): { values?: any; error?: string } => {
  try {
    const triggerResponseJson = parse(data?.yaml || '')
    triggerResponseJson.trigger.enabled = enabled
    return { values: triggerResponseJson.trigger }
  } catch (e) {
    return { error: getString('pipeline.triggers.cannotParseTriggersData') }
  }
}
const TriggerStatus = {
  FAILED: 'FAILED',
  UNKNOWN: 'UNKNOWN',
  ERROR: 'ERROR',
  TIMEOUT: 'TIMEOUT',
  UNAVAILABLE: 'UNAVAILABLE',
  SUCCESS: 'SUCCESS'
}

export const errorStatusList = [
  TriggerStatus.FAILED,
  TriggerStatus.UNKNOWN,
  TriggerStatus.ERROR,
  TriggerStatus.TIMEOUT,
  TriggerStatus.UNAVAILABLE
]
