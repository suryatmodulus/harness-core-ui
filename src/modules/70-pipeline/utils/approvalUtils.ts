/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { ApprovalInstanceResponse } from 'services/pipeline-ng'

export type ApprovalStatusType = ApprovalInstanceResponse['status']

export const ApprovalStatus: Record<ApprovalStatusType, ApprovalStatusType> = {
  WAITING: 'WAITING',
  APPROVED: 'APPROVED',
  REJECTED: 'REJECTED',
  FAILED: 'FAILED',
  EXPIRED: 'EXPIRED'
}

export function isApprovalWaiting(status: ApprovalStatusType): boolean {
  return status === ApprovalStatus.WAITING
}

export function isApprovalApproved(status: ApprovalStatusType): boolean {
  return status === ApprovalStatus.APPROVED
}

export function isApprovalRejected(status: ApprovalStatusType): boolean {
  return status === ApprovalStatus.REJECTED
}

export function isApprovalFailed(status: ApprovalStatusType): boolean {
  return status === ApprovalStatus.FAILED
}

export function isApprovalExpired(status: ApprovalStatusType): boolean {
  return status === ApprovalStatus.EXPIRED
}
