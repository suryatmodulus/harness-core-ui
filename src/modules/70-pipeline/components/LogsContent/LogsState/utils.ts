/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { escapeStringRegexp } from '@common/utils/StringUtils'

import type { State } from './types'

export function getRegexForSearch(str: string): RegExp {
  return new RegExp(escapeStringRegexp(str), 'gi')
}

export function getDefaultReducerState(props: Partial<Pick<State, 'selectedStage' | 'selectedStep'>> = {}): State {
  return {
    units: [],
    logKeys: [],
    selectedStage: props.selectedStage || '',
    selectedStep: props.selectedStep || '',
    dataMap: {},
    searchData: {
      text: '',
      currentIndex: 0,
      linesWithResults: []
    }
  }
}

export const logsCache = new Map<string, string>()
