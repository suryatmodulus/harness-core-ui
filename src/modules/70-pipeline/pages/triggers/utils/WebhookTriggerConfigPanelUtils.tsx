/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { SetStateAction, Dispatch } from 'react'
import cx from 'classnames'
import { FormInput, SelectOption, Text, Container, FontVariation, Color } from '@wings-software/uicore'
import { isEmpty, isUndefined } from 'lodash-es'
import type { StringKeys, UseStringsReturn } from 'framework/strings'
import { GitSourceProviders } from './TriggersListUtils'
import { ConnectorSection } from '../views/ConnectorSection'
import { eventTypes } from './TriggersWizardPageUtils'
import css from '../views/WebhookTriggerConfigPanel.module.scss'

export const autoAbortPreviousExecutionsTypes = [
  eventTypes.PUSH,
  eventTypes.PULL_REQUEST,
  eventTypes.ISSUE_COMMENT,
  eventTypes.MERGE_REQUEST
]

export const getAutoAbortDescription = ({
  event,
  getString
}: {
  event: string
  getString: (key: StringKeys) => string
}): string => {
  if (event === eventTypes.PUSH) {
    return getString('pipeline.triggers.triggerConfigurationPanel.autoAbortPush')
  } else if (event === eventTypes.PULL_REQUEST || event === eventTypes.MERGE_REQUEST) {
    return getString('pipeline.triggers.triggerConfigurationPanel.autoAbortPR')
  } else if (event === eventTypes.ISSUE_COMMENT) {
    return getString('pipeline.triggers.triggerConfigurationPanel.autoAbortIssueComment')
  }
  return ''
}

export const handleSourceRepoChange = ({ e, formikProps }: { e: SelectOption; formikProps: any }): void => {
  if (e.value === GitSourceProviders.CUSTOM.value) {
    formikProps.setValues({
      ...formikProps.values,
      sourceRepo: e.value,
      connectorRef: undefined,
      repoName: '',
      actions: undefined,
      anyAction: false,
      secretToken: undefined
    })
  } else {
    formikProps.setValues({
      ...formikProps.values,
      sourceRepo: e.value,
      connectorRef: undefined,
      repoName: '',
      actions: undefined,
      anyAction: false,
      secretToken: undefined,
      headerConditions: undefined
    })
  }
}

export const renderNonCustomEventFields = ({
  sourceRepo,
  formikProps,
  event,
  eventOptions,
  getString,
  actionsOptions,
  actions
}: {
  sourceRepo?: string
  formikProps: any
  event?: string
  eventOptions: SelectOption[]
  getString: UseStringsReturn['getString']
  actionsOptions: SelectOption[]
  actions: SelectOption[]
}): JSX.Element => {
  return (
    <>
      {sourceRepo && <ConnectorSection formikProps={formikProps} />}
      <FormInput.Select
        className={cx(!event && css.bottomMarginZero)}
        key={event}
        label={getString('pipeline.triggers.triggerConfigurationPanel.event')}
        name="event"
        placeholder={getString('pipeline.triggers.triggerConfigurationPanel.eventPlaceholder')}
        items={eventOptions}
        onChange={e => {
          const additionalValues: any = {}

          if (e.value === eventTypes.PUSH) {
            additionalValues.sourceBranchOperator = undefined
            additionalValues.sourceBranchValue = undefined
          }

          if (e.value !== eventTypes.TAG) {
            additionalValues.tagConditionOperator = undefined
            additionalValues.tagConditionValue = undefined
          }

          formikProps.setValues({
            ...formikProps.values,
            event: e.value,
            actions: e.value === eventTypes.PUSH ? [] : undefined,
            anyAction: false,
            ...additionalValues
          })
        }}
      />
      {event && (
        <>
          {actionsOptions?.length !== 0 && (
            <div className={css.actionsContainer}>
              <div>
                <Text
                  font={{ variation: FontVariation.FORM_INPUT_TEXT, weight: 'semi-bold' }}
                  color={Color.GREY_600}
                  style={{ marginBottom: 'var(--spacing-xsmall)' }}
                >
                  {getString('pipeline.triggers.triggerConfigurationPanel.actions')}
                </Text>
                <FormInput.MultiSelect
                  className={css.multiSelect}
                  name="actions"
                  items={actionsOptions}
                  // yaml design: empty array means selecting all
                  disabled={Array.isArray(actions) && isEmpty(actions)}
                  onChange={e => {
                    if (!e || (Array.isArray(e) && isEmpty(e))) {
                      formikProps.setFieldValue('actions', undefined)
                    } else {
                      formikProps.setFieldValue('actions', e)
                    }
                  }}
                />
              </div>
              <Container className={css.anyActionContainer}>
                <FormInput.CheckBox
                  name="anyAction"
                  key={Date.now()}
                  label={getString('pipeline.triggers.triggerConfigurationPanel.anyActions')}
                  defaultChecked={(Array.isArray(actions) && actions.length === 0) || false}
                  className={css.checkboxAlignment}
                  onClick={(e: React.FormEvent<HTMLInputElement>) => {
                    formikProps.setFieldTouched('actions', true)
                    if (e.currentTarget?.checked) {
                      formikProps.setFieldValue('actions', [])
                    } else {
                      formikProps.setFieldValue('actions', undefined)
                    }
                  }}
                />
              </Container>
            </div>
          )}
          {autoAbortPreviousExecutionsTypes.includes(event) && (
            <>
              <FormInput.CheckBox
                style={{ position: 'relative', left: '0' }}
                name="autoAbortPreviousExecutions"
                label="Auto-abort Previous Execution"
                className={css.checkboxAlignment}
              />
              <Text className={css.autoAbortDescription}>{getAutoAbortDescription({ event, getString })}</Text>
            </>
          )}
        </>
      )}
    </>
  )
}
interface ActionsOptionsMapInterface {
  [key: string]: string[]
}

export const getEventLabelMap = (event: string): string => {
  // add space between camelcase-separated words
  if (event === eventTypes.PULL_REQUEST) {
    return 'Pull Request'
  } else if (event === eventTypes.MERGE_REQUEST) {
    return 'Merge Request'
  } else if (event === eventTypes.ISSUE_COMMENT) {
    return 'Issue Comment'
  }
  return event
}

export const getEventAndActions = ({
  data,
  sourceRepo
}: {
  data: {
    [key: string]: {
      [key: string]: string[]
    }
  }
  sourceRepo: string
}): { eventOptions: SelectOption[]; actionsOptionsMap: ActionsOptionsMapInterface } => {
  const filteredData = data?.[sourceRepo] || {}
  const eventOptions = Object.keys(filteredData).map(event => ({
    label: getEventLabelMap(event),
    value: event
  }))
  return { eventOptions, actionsOptionsMap: filteredData }
}

export const clearEventsAndActions = ({
  newActionsOptionsMap,
  formikProps,
  setActionsOptions,
  event
}: {
  newActionsOptionsMap: ActionsOptionsMapInterface
  formikProps: any
  setActionsOptions: Dispatch<SetStateAction<SelectOption[]>>
  event: string
}): void => {
  if (isUndefined(newActionsOptionsMap[event])) {
    formikProps.setFieldValue('event', undefined)
    return
  }

  const newActionsOptions = newActionsOptionsMap[event]?.map((val: string) => ({ label: val, value: val }))
  setActionsOptions(newActionsOptions)
  if (newActionsOptions?.length === 0) {
    formikProps.setFieldValue('actions', [])
  }
}
