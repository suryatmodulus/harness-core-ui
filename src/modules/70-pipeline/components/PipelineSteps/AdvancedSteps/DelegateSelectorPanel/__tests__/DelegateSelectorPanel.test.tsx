/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { fireEvent, getByText, render, waitFor } from '@testing-library/react'
import { Formik, FormikForm } from '@wings-software/uicore'
import { TestWrapper } from '@common/utils/testUtils'
import { accountPathProps } from '@common/utils/routeUtils'
import routes from '@common/RouteDefinitions'
import DelegateSelectorPanel from '../DelegateSelectorPanel'

const mockGetCallFunction = jest.fn()
jest.mock('services/portal', () => ({
  useGetDelegateSelectorsUpTheHierarchy: jest.fn().mockImplementation(args => {
    mockGetCallFunction(args)
    return []
  })
}))

const params = {
  accountId: 'testAcc'
}
const TEST_PATH = routes.toDelegateList({
  ...accountPathProps
})

function WrapperComponent(props: { initialValues: any }): JSX.Element {
  const { initialValues } = props
  return (
    <TestWrapper path={TEST_PATH} pathParams={params}>
      <Formik
        enableReinitialize={true}
        initialValues={initialValues}
        onSubmit={jest.fn()}
        formName="delegateSelectorPanelForm"
      >
        {formikProps => {
          return (
            <FormikForm>
              <DelegateSelectorPanel formikProps={formikProps} isReadonly={false} />
            </FormikForm>
          )
        }}
      </Formik>
    </TestWrapper>
  )
}

describe('<DelegateSelectorPanel /> test', () => {
  test('snapshot testing', () => {
    const { container } = render(
      <WrapperComponent initialValues={{ delegateSelectors: ['harness-sample-k8s-delegate'] }} />
    )
    expect(container).toMatchSnapshot()
  })

  test('should be able to select multiple tags from delegate selector', () => {
    const { container } = render(
      <WrapperComponent initialValues={{ delegateSelectors: ['harness-sample-k8s-delegate'] }} />
    )
    const selectedTag = getByText(container, 'harness-sample-k8s-delegate')
    expect(selectedTag).not.toBeNull()
    const inputBox = container.getElementsByClassName('bp3-multi-select-tag-input-input')[0]
    fireEvent.change(inputBox, { target: { value: 'new-tag' } })
    const overlay = container.getElementsByClassName('bp3-transition-container')[0]
    expect(overlay).not.toBeUndefined()
    const option = getByText(container, 'Create "new-tag"')
    expect(container).toMatchSnapshot()
    waitFor(() => fireEvent.click(option))
  })
})
