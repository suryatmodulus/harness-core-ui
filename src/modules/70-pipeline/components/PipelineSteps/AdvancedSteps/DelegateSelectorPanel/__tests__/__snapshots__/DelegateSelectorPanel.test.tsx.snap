// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DelegateSelectorPanel /> test should be able to select multiple tags from delegate selector 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      class="FormikForm--main"
    >
      <div>
        <div
          class="bp3-form-group"
          data-id="undefined-0"
        >
          <div
            class="bp3-form-content"
          >
            <div
              class="Layout--horizontal StyledProps--main MultiTypeInput--main"
              style="flex-grow: 1;"
            >
              <div
                class="wrapper wrapper"
                data-name="DelegateSelectors"
              >
                <div
                  class="bp3-popover-wrapper delegateInput delegatePopover bp3-fill"
                >
                  <div
                    class="bp3-popover-target bp3-popover-open"
                  >
                    <div
                      class=""
                    >
                      <div
                        class="bp3-input bp3-tag-input bp3-fill bp3-multi-select"
                      >
                        <div
                          class="bp3-tag-input-values"
                        >
                          <span
                            class="bp3-tag bp3-intent-danger bp3-minimal TagInput--tag"
                            data-tag-index="0"
                          >
                            <span
                              class="bp3-text-overflow-ellipsis bp3-fill"
                            >
                              harness-sample-k8s-delegate
                            </span>
                            <button
                              class="bp3-tag-remove"
                              type="button"
                            >
                              <span
                                class="bp3-icon bp3-icon-small-cross"
                                icon="small-cross"
                              >
                                <svg
                                  data-icon="small-cross"
                                  height="16"
                                  viewBox="0 0 16 16"
                                  width="16"
                                >
                                  <desc>
                                    small-cross
                                  </desc>
                                  <path
                                    d="M9.41 8l2.29-2.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-2.29-2.3a1.003 1.003 0 00-1.42 1.42L6.59 8 4.3 10.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
                                    fill-rule="evenodd"
                                  />
                                </svg>
                              </span>
                            </button>
                          </span>
                          <input
                            class="bp3-input-ghost bp3-multi-select-tag-input-input"
                            value="new-tag"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="bp3-overlay bp3-overlay-open bp3-overlay-inline"
                  >
                    <div
                      class="bp3-transition-container bp3-popover-appear bp3-popover-appear-active"
                      style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
                    >
                      <div
                        class="bp3-popover bp3-minimal bp3-multi-select-popover"
                      >
                        <div
                          class="bp3-popover-content"
                        >
                          <div>
                            <ul
                              class="bp3-menu"
                            >
                              <li
                                class=""
                              >
                                <a
                                  class="bp3-menu-item bp3-popover-dismiss"
                                >
                                  <span
                                    class="bp3-icon bp3-icon-add"
                                    icon="add"
                                  >
                                    <svg
                                      data-icon="add"
                                      height="16"
                                      viewBox="0 0 16 16"
                                      width="16"
                                    >
                                      <desc>
                                        add
                                      </desc>
                                      <path
                                        d="M10.99 6.99h-2v-2c0-.55-.45-1-1-1s-1 .45-1 1v2h-2c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1v-2h2c.55 0 1-.45 1-1s-.45-1-1-1zm-3-7c-4.42 0-8 3.58-8 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.68 6-6 6z"
                                        fill-rule="evenodd"
                                      />
                                    </svg>
                                  </span>
                                  <div
                                    class="bp3-text-overflow-ellipsis bp3-fill"
                                  >
                                    Create "new-tag"
                                  </div>
                                </a>
                              </li>
                            </ul>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span
                class="bp3-popover-wrapper MultiTypeInput--wrapper"
              >
                <span
                  class="bp3-popover-target"
                >
                  <button
                    class="MultiTypeInput--btn MultiTypeInput--FIXED"
                  >
                    <span
                      class="bp3-icon StyledProps--main"
                      data-icon="fixed-input"
                    >
                      <svg
                        height="12"
                        viewBox="0 0 9 9"
                        width="12"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </button>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`<DelegateSelectorPanel /> test snapshot testing 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      class="FormikForm--main"
    >
      <div>
        <div
          class="bp3-form-group"
          data-id="undefined-0"
        >
          <div
            class="bp3-form-content"
          >
            <div
              class="Layout--horizontal StyledProps--main MultiTypeInput--main"
              style="flex-grow: 1;"
            >
              <div
                class="wrapper wrapper"
                data-name="DelegateSelectors"
              >
                <div
                  class="bp3-popover-wrapper delegateInput delegatePopover bp3-fill"
                >
                  <div
                    class="bp3-popover-target"
                  >
                    <div
                      class=""
                    >
                      <div
                        class="bp3-input bp3-tag-input bp3-fill bp3-multi-select"
                      >
                        <div
                          class="bp3-tag-input-values"
                        >
                          <span
                            class="bp3-tag bp3-intent-danger bp3-minimal TagInput--tag"
                            data-tag-index="0"
                          >
                            <span
                              class="bp3-text-overflow-ellipsis bp3-fill"
                            >
                              harness-sample-k8s-delegate
                            </span>
                            <button
                              class="bp3-tag-remove"
                              type="button"
                            >
                              <span
                                class="bp3-icon bp3-icon-small-cross"
                                icon="small-cross"
                              >
                                <svg
                                  data-icon="small-cross"
                                  height="16"
                                  viewBox="0 0 16 16"
                                  width="16"
                                >
                                  <desc>
                                    small-cross
                                  </desc>
                                  <path
                                    d="M9.41 8l2.29-2.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-2.29-2.3a1.003 1.003 0 00-1.42 1.42L6.59 8 4.3 10.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
                                    fill-rule="evenodd"
                                  />
                                </svg>
                              </span>
                            </button>
                          </span>
                          <input
                            class="bp3-input-ghost bp3-multi-select-tag-input-input"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span
                class="bp3-popover-wrapper MultiTypeInput--wrapper"
              >
                <span
                  class="bp3-popover-target"
                >
                  <button
                    class="MultiTypeInput--btn MultiTypeInput--FIXED"
                  >
                    <span
                      class="bp3-icon StyledProps--main"
                      data-icon="fixed-input"
                    >
                      <svg
                        height="12"
                        viewBox="0 0 9 9"
                        width="12"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </button>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
